tests/functional/codegen/features/iteration/test_for_range.py

```diff
@@ -271,17 +273,21 @@ def test():
 
 
 @pytest.mark.parametrize("typ", ["uint8", "int128", "uint256"])
-def test_for_range_oob_check(get_contract, tx_failed, typ):
+def test_for_range_oob_check(get_contract, tx_failed, typ, experimental_codegen):
     code = f"""
 @external
 def test():
     x: {typ} = max_value({typ})
     for i: {typ} in range(x, x + 2, bound=2):
         pass
     """
-    c = get_contract(code)
-    with tx_failed():
-        c.test()
+    if experimental_codegen:
+        with pytest.raises(StaticAssertionException):
```

Comment:
this is so cool

Reply:
we might want to keep showing that there is a runtime check though. can we pass `x` as a parameter thru calldata?